### PR TITLE
report drcluster deploy status using manifestwork

### PR DESCRIPTION
**Pull request by rakeshgm**
Read DRCluster manifest-work status and report if not successful and fix unit tests intermittent failures on an update of an object.

### CODE REVIEW

**Comment by rakeshgm** [comment]
i still need to fix the unit test failures.

File: controllers/drcluster_controller.go [hunk 1]
```diff
@@ -124,6 +124,11 @@ func (r *DRClusterReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 		return ctrl.Result{}, fmt.Errorf("drclusters deploy: %w", u.validatedSetFalseAndUpdate("DrClustersDeployFailed", err))
 	}
 
+	deployed := u.getDRClusterDeployedStatus(drcluster)
+	if !deployed {
+		return ctrl.Result{}, fmt.Errorf("drcluster deploy failed")
```
Comment (rakeshgm) [inline controllers/drcluster_controller.go:129]: I think the error message can be improved. thoughts ?

**Comment by rakeshgm** [comment]
> i still need to fix the unit test failures.
I have fixed the units tests.

**Comment by rakeshgm** [comment]
> i still need to fix the unit test failures.

unit tests are passing now.

File: controllers/drcluster_controller.go [hunk 2]
```diff
@@ -370,6 +376,24 @@ func (u *drclusterInstance) initializeStatus() {
 	}
 }
 
+func (u *drclusterInstance) getDRClusterDeployedStatus(drcluster *ramen.DRCluster) error {
+	mw, err := u.mwUtil.GetDrClusterManifestWork(drcluster.Name)
```
Comment (ShyamsundarR) [inline controllers/drcluster_controller.go:380]: The errors here appear in the resource conditions status, we can possibly word these better, some suggestions:

Line 381: "error fetching DRCluster ManifestWork"
Line 386: "missing DRCluster ManifestWork resource"
Line 391: "DRCluster ManifestWork is not in applied state"

Comment (rakeshgm) [reply controllers/drcluster_controller.go:380]: error messages fixed.

**Review by ShyamsundarR — APPROVED** [review]

**Pull request merged**